synapse/handlers/sync.py

```diff
@@ -1553,16 +1553,17 @@ def _calculate_state(
     # as we may not have sent them to the client before.  We find these membership
     # events by filtering them out of timeline_start, which has already been filtered
     # to only include membership events for the senders in the timeline.
+    # In practice, we can do this by removing them from the p_ids list.
+    # see https://github.com/matrix-org/synapse/pull/2970
```

Comment:
I'm not sure a link that you have to reconstruct which proves that this implementation is equivalent to some other implementation that never landed is terribly helpful.

can we just say something like "In practice, we do this by removing them from p_ids, which is the list of things we know we have already sent to the client." ?